umake/frameworks/ide.py

```diff
@@ -473,3 +474,83 @@ def post_install(self):
                                                      categories=categories))
         if not self.was_in_arduino_group:
             UI.delayed_display(DisplayMessage(_("You need to logout and login again for your installation to work")))
+            
+class BaseNetBeans(umake.frameworks.baseinstaller.BaseInstaller):
+    """The base for all Netbeans installers."""
+    
+    BASE_URL = "http://download.netbeans.org/netbeans/8.0.2/final/"
+
+    def __init__(self, category, version="8.0.2", flavour=""):
+        #add a separator to the string, like -cpp
+        if flavour != "":
+            flavour = '-' + flavour
+        super().__init__(name="Netbeans",
+                         description=_("Netbeans IDE " + version),
+                         category=category, 
+                         only_on_archs=['i386', 'amd64', 'arm'],
+                         download_page=None,
+                         dir_to_decompress_in_tarball="netbeans" + flavour,
+                         desktop_filename="netbeans" + flavour + ".desktop",
+                         packages_requirements=['openjdk-7-jdk', 'jayatana'])
+        
+        self.url_string = self.search_url_string(self, version, flavour)
+        data = self.url_string.split(", ")
+        self.url_suffix = data[0]
+        self.md5sum = data[2]
+        self.complete_name = "netbeans" + flavour
+    
+    @property
+    def search_url_string(self, version_number, flavour=""):
+      #connect to the page and look for the version
+      data = urlopen("https://netbeans.org/images_www/v6/download/" \
```

Comment:
Please do not use urlopen() from anything that is in in __init__(). This would mean that everytime you:
umake --help
or even:
umake <tab><tab>, you are going to run that request. I think you should only do that in the first step of installation (so, from download_provider_page) and download it asynchronously through the DownloadCenter (to not block the UI on long request).